code/datums/smooth.dm

```diff
@@ -0,0 +1,206 @@
+			holder.overlays += top_right
+			holder.overlays += bottom_right
+			holder.overlays += bottom_left
+
+
+/datum/tile_smoother/proc/make_nw_corner()
+	var/sdir = ""
+	if((cardinal_adj & NORTH) && (cardinal_adj & WEST))
+		if(NORTHWEST in diagonal_adj)
+			sdir = "f"
+		else
+			sdir = "nw"
+	else
+		if(cardinal_adj & NORTH)
+			sdir = "n"
+		else if(cardinal_adj & WEST)
+			sdir = "w"
+		else
+			sdir = "i"
+	top_left = image(holder.icon, "1-[sdir]")
+
+/datum/tile_smoother/proc/make_ne_corner()
+	var/sdir = ""
+	if((cardinal_adj & NORTH) && (cardinal_adj & EAST))
+		if(NORTHEAST in diagonal_adj)
+			sdir = "f"
+		else
+			sdir = "ne"
+	else
+		if(cardinal_adj & NORTH)
+			sdir = "n"
+		else if(cardinal_adj & EAST)
+			sdir = "e"
+		else
+			sdir = "i"
+	top_right = image(holder.icon, "2-[sdir]")
+
+/datum/tile_smoother/proc/make_sw_corner()
+	var/sdir = ""
+	if((cardinal_adj & SOUTH) && (cardinal_adj & WEST))
+		if(SOUTHWEST in diagonal_adj)
+			sdir = "f"
+		else
+			sdir = "sw"
+	else
+		if(cardinal_adj & SOUTH)
+			sdir = "s"
+		else if(cardinal_adj & WEST)
+			sdir = "w"
+		else
+			sdir = "i"
+	bottom_left = image(holder.icon, "3-[sdir]")
+
+/datum/tile_smoother/proc/make_se_corner()
+	var/sdir = ""
+	if((cardinal_adj & SOUTH) && (cardinal_adj & EAST))
+		if(SOUTHEAST in diagonal_adj)
+			sdir = "f"
+		else
+			sdir = "se"
+	else
+		if(cardinal_adj & SOUTH)
+			sdir = "s"
+		else
+			if(cardinal_adj & EAST)
+				sdir = "e"
+			else
+				sdir = "i"
+	bottom_right = image(holder.icon, "4-[sdir]")
+
+/datum/tile_smoother/proc/update_neighbors()
+	for(var/atom/A in orange(1,holder))
+		if(A.smoother)
+			A.smoother.smooth()
+
+/proc/find_type_in_direction(atom/source, direction, list/siblings=null, range=1)
+	var/x_offset = 0
+	var/y_offset = 0
+
+	if(direction & NORTH)
+		y_offset = range
+	else if(direction & SOUTH)
+		y_offset -= range
+
+	if(direction & EAST)
+		x_offset = range
+	else if(direction & WEST)
+		x_offset -= range
+
+	if(istype(source, /turf))
+		if(siblings)
+			for(var/a_type in siblings)
+				if(ispath(a_type, /obj))
+					var/atom/A = locate(a_type) in locate(source.x + x_offset, source.y + y_offset, source.z)
+					if(A && A.type == a_type)
+						return A
+				else
+					var/turf/T = locate(source.x + x_offset, source.y + y_offset, source.z)
+					if(T.type == a_type)
+						return T
+			return null
+		var/turf/T = locate(source.x + x_offset, source.y + y_offset, source.z)
+		return T.type == source.type ? T : null
+
+	if(siblings)
+		for(var/a_type in siblings)
+			if(ispath(a_type, /turf))
+				var/turf/T = locate(source.x + x_offset, source.y + y_offset, source.z)
+				if(T.type == a_type)
+					return T
+			else
+				var/atom/A = locate(a_type) in locate(source.x + x_offset, source.y + y_offset, source.z)
+				if(A && A.type == a_type)
+					return A
+		return null
+	var/atom/A = locate(source.type) in locate(source.x + x_offset, source.y + y_offset, source.z)
+	return A && A.type == source.type ? A : null
```

Comment:
I'm not saying you should change this, but you could get the same effect with return A && A.type == source.type && A